db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/FullOuterJoinNode.java

```diff
@@ -0,0 +1,248 @@
+        leftFromTable.pushExpressions(noPredicates);
+        rightFromTable.pushExpressions(noPredicates);
+    }
+
+    /**
+     * Generate the code for an full outer join node.
+     *
+     * @throws StandardException Thrown on error
+     */
+    @Override
+    public void generate(ActivationClassBuilder acb, MethodBuilder mb) throws StandardException{
+        super.generateCore(acb,mb,FULLOUTERJOIN);
+    }
+
+    @Override
+    protected int addOuterJoinArguments(ActivationClassBuilder acb, MethodBuilder mb) throws StandardException{
+		/* Nulls from the left */
+        leftResultSet.getResultColumns().generateNulls(acb,mb);
+
+        /* Nulls from the right */
+        rightResultSet.getResultColumns().generateNulls(acb,mb);
+
+		/* wasRightOuterJoin */
+        mb.push(false);
+
+        return 3;
+    }
+
+    /**
+     * Return the number of arguments to the join result set.
+     */
+    @Override
+    protected int getNumJoinArguments(){
+		/* We add two more arguments than the superclass does */
+        return super.getNumJoinArguments()+3;
+    }
+
+    @Override
+    public String printExplainInformation(String attrDelim, int order) throws StandardException {
+        JoinStrategy joinStrategy = RSUtils.ap(this).getJoinStrategy();
+        StringBuilder sb = new StringBuilder();
+        sb.append(spaceToLevel())
+                .append(joinStrategy.getJoinStrategyType().niceName()).append("FullOuter").append("Join(")
+                .append("n=").append(order)
+                .append(attrDelim).append(getFinalCostEstimate(false).prettyProcessingString(attrDelim));
+        if (joinPredicates !=null) {
+            List<String> joinPreds = Lists.transform(PredicateUtils.PLtoList(joinPredicates), PredicateUtils.predToString);
+            if (!joinPreds.isEmpty()) {
+                sb.append(attrDelim).append("preds=[").append(Joiner.on(",").skipNulls().join(joinPreds)).append("]");
+            }
+        }
+        sb.append(")");
+        return sb.toString();
+    }
+
+    @Override
+    public FromTable transformOuterJoins(ValueNode predicateTree,int numTables) throws StandardException{
+
+        super.transformOuterJoins(null,numTables);
+        if(PredicateSimplificationVisitor.isBooleanTrue(joinClause)){
+            JoinNode ij=(JoinNode)
+                    getNodeFactory().getNode(
+                            C_NodeTypes.JOIN_NODE,
+                            leftResultSet,
+                            rightResultSet,
+                            joinClause,
+                            null,
+                            resultColumns,
+                            null,
+                            null,
+                            getContextManager());
+            ij.setTableNumber(tableNumber);
+            ij.setSubqueryList(subqueryList);
+            ij.setAggregateVector(aggregateVector);
+
+            leftResultSet.notFlattenableJoin();
+            rightResultSet.notFlattenableJoin();
```

Comment:
I think you are right. I add the two lines just to make the situation simpler. Let me relax and test it out.

Reply:
I removed the two lines above, so that the left and right sources can be flattened to the main SELECT query. Given the following query:
```
select * from t1 inner join t2 on a1=a2 full join t3 on 1=1;
```
All 3 tables t1, t2 and t3 will be pulled into the main SELECT query's FROM table list.